include/mbedtls/config_psa.h

```diff
@@ -308,6 +308,11 @@ extern "C" {
 #define PSA_HAVE_SOFT_BLOCK_AEAD 1
 #endif
 
+#if defined(PSA_WANT_ALG_ECDSA) && defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)
+#define PSA_HAS_FULL_ECDSA 1
```

Comment:
My first reaction was "can we really define macros in the `PSA_` namespace like that? Isn't that namespace reserved for the PSA spec?" then I saw the context that already defines lots of such macros, so apparently it's OK.

So instead, I'll just ask that for consistency with the surrounding macros: `PSA_HAVE` not `PSA_HAS` :)

Reply:
> then I saw the context that already defines lots of such macros

eheheheh I just thought that I needed a macro like that one for better readability of the code so I went looking for "similar" ideas in the main header files and I found those ones :)
Of course your suggestion is absolutely right!

Since the branch has to be rebased anyway (due to the dependency from #7082) if you don't mind I would like to wait for the CI to complete before applying this change. I propose this way because this morning the CI was complaining for a conflict in `ssl_tls.c` for this PR. I tried to verify it locally by trying to merge into `development`, but I didn't get the same problem. Therefore I rebased on #7082 (which I know that is not aligned to development, but it didn't suffer from this problem this morning) to see if the conflict is still there

Reply:
I just merged #7082 and the CI here came back green (after only 4h, which these days is pretty good I guess), so I think you can now rebase - and fix the typo that was left out of #7082.

Reply:
> can we really define macros in the PSA_ namespace like that? Isn't that namespace reserved for the PSA spec?

This is a long-time-unresolved issue: we don't know what prefix to use for things that are not standard PSA definitions, but are not related to Mbed TLS legacy crypto APIs or Mbed TLS X.509 and TLS APIs.

Using a `PSA/psa` prefix for nonstandard stuff is acceptable as long as we ensure that it never conflicts with an official PSA definition. So if PSA adopts one of those names, we'll have to rename our own. And if we make those a stable interface, we rely on our ability to influence PSA specifications. Fortunately that's rarely the case — most of these names are either not part our API or designated as experimental — but a few names have crept up (in `psa/crypto_extra.h`) which are stable Mbed TLS APIs that were on PSA standardization track but PSA didn't adopt the feature.